### PR TITLE
Push recipient

### PR DESCRIPTION
**Pull request by Kate5**
Send error message if DeliveryOptions.setBroadcast() or DeliveryOptions.setSinglecast() is not called
(BKNDLSS-8533)

### CODE REVIEW

File: src/com/backendless/Persistence.java [hunk 1]
```diff
@@ -788,7 +788,7 @@ private void checkPageSizeAndOffset( BackendlessDataQuery dataQuery ) throws Bac
 
   public static String getSimpleName( Class clazz )
   {
-    if( clazz.equals( BackendlessUser.class ) )
+    if( clazz == BackendlessUser.class )
```
Comment (markpiller) [inline src/com/backendless/Persistence.java]: Why do we need to check equality like this?

File: src/com/backendless/exceptions/ExceptionMessage.java [hunk 2]
```diff
@@ -88,7 +88,7 @@
 
   public final static String WRONG_SUBSCRIPTION_STATE = "cannot resume a subscription, which is not paused.";
   public final static String WRONG_EXPIRATION_DATE = "Wrong expiration date";
-
+  public final static String RECIPIENT_MISSING = "Push notification recipient is missing: call DeliveryOptions.setPushBroadcast() or DeliveryOptions.setPushSinglecast()";
```
Comment (markpiller) [inline src/com/backendless/exceptions/ExceptionMessage.java]: Change the message to:

"Push notification recipient is missing: use either DeliveryOptions.setPushBroadcast() or DeliveryOptions.setPushSinglecast() to set the recipients"

File: src/com/backendless/persistence/BackendlessSerializer.java [hunk 3]
```diff
@@ -26,9 +47,20 @@
...
+  private static Object serializeToMap( Object entity,  Map<Object, Map<String, Object>> serializedCache )
+  {
+    if(entity.getClass().isArray())
+    {
+      return serializeArray( entity, serializedCache );
+    }
+
     Map<String, Object> serializedEntity = new HashMap<String, Object>();
 
-    if( entity.getClass().equals( BackendlessUser.class ) )
```
Comment (markpiller) [inline src/com/backendless/persistence/BackendlessSerializer.java]: why??

File: src/com/backendless/Persistence.java (hunk 1 above)
Comment (ksv510) [inline src/com/backendless/Persistence.java]: clazz.equals( BackendlessUser.class ) the same as clazz == BackendlessUser.class

File: src/com/backendless/persistence/BackendlessSerializer.java (hunk 3 above)
Comment (ksv510) [inline src/com/backendless/persistence/BackendlessSerializer.java]: clazz.equals( BackendlessUser.class ) the same as clazz == BackendlessUser.class

**Pull request closed without merging**